src/cityreader.py

```diff
@@ -48,4 +60,25 @@
 # Tucson: (32.1558,-110.8777)
 # Salt Lake City: (40.7774,-111.9301)
 
-# TODO
\ No newline at end of file
+lat1 = input("Enter lat1:")
+lon1 = input("Enter lon1:")
+lat2 = input("Enter lat2:")
+lon2 = input("Enter lon2:")
+
+if lat2 < lat1:
+    lat1, lat2 = lat2, lat1
+
+if lon2 < lon1:
+    lon1, lon2 = lon2, lon1
+
+# This is broken
+within = []
+for c in cities:
```

Comment:
Not completely broken as some results are returned properly so good job on tackling the stretch goal.